### ISSUE TITLE
🐛 `FileNotFoundError` / `OSError`

### ISSUE THREAD

**Issue by suezaan** [FCP-INDI/C-PAC#1752]
### Describe the bug

'Excessive usage report' problem when I try to run single subject data (2 movie watching runs, 2 rest runs) for a test

### To reproduce

1. Following default_pipeline.yml

 I have created data_settings & data_configuration file and try to run a default pipeline.yml with the following command:

```
docker run --rm \
	      -u $UID:$UID \
        -v /Users/hal/Desktop/test:/bids_dataset \
        -v /Users/hal/Desktop/test:/outputs \
        -v /tmp:/tmp \
        -v /Users/hal/Desktop/test:/configs \
        -v /Users/hal/Desktop/test:/resources \
        fcpindi/c-pac:latest /bids_dataset /outputs participant --pipeline_file /outputs/default_pipeline.yml --data_config_file /outputs/data_config_cpac.yml
```

2. Error message 
Everytime I run this, I get this error: 
```
220629-04:28:40,421 nipype.workflow WARNING:
	 Excessive usage report failed for /outputs/log/pipeline_analysis/sub-NDARAB653ZXP_ses-1/callback.log
Report generation failed for /outputs/log/pipeline_analysis/sub-NDARAB653ZXP_ses-1/callback.log
NoneType: None
220629-04:28:40,425 nipype.workflow INFO:
	 

Error of subject workflow cpac_sub-NDARAB653ZXP_ses-1

CPAC run error:

    Pipeline configuration: analysis
    Subject workflow: cpac_sub-NDARAB653ZXP_ses-1
    Elapsed run time (minutes): 3.4772278587023417
    Timing information saved in /outputs/log/cpac_individual_timing_analysis.csv
    System time of start:      2022-06-29 04:25:14


220629-04:28:40,430 nipype.workflow INFO:
	 Removing working dir: /tmp/cpac_sub-NDARAB653ZXP_ses-1
Traceback (most recent call last):
  File "/code/CPAC/pipeline/cpac_pipeline.py", line 515, in run_workflow
    workflow.run(plugin=plugin, plugin_args=plugin_args)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/workflows.py", line 632, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/linear.py", line 70, in run
    report_nodes_not_run(notrun)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/tools.py", line 98, in report_nodes_not_run
    ("Workflow did not execute cleanly. " "Check log for details")
RuntimeError: Workflow did not execute cleanly. Check log for details
```

and here is the log files 
[callback.log](https://github.com/FCP-INDI/C-PAC/files/9007282/callback.log)
[pypeline.log](https://github.com/FCP-INDI/C-PAC/files/9007283/pypeline.log)

As a side note, I am currently running preprocessing with frmriprep in the same device and this, too seems like it's having problem with memory usage (it fails with BrokenProcessPool). I wonder if this is related to the similar memory usage problem in cpac. 

### Preconfig

- [X] default
- [ ] abcd-options
- [ ] anat-only
- [ ] blank
- [ ] ccs-options
- [ ] fmriprep-options
- [ ] fx-options
- [ ] monkey
- [ ] monkey-ABCD
- [ ] ndmg
- [ ] nhp-macaque
- [ ] preproc
- [ ] rbc-options
- [ ] rodent

### Run command

`docker run --rm \
	      -u $UID:$UID \
        -v /Users/hal/Desktop/test:/bids_dataset \
        -v /Users/hal/Desktop/test:/outputs \
        -v /tmp:/tmp \
        -v /Users/hal/Desktop/test:/configs \
        -v /Users/hal/Desktop/test:/resources \
        fcpindi/c-pac:latest /bids_dataset /outputs participant --pipeline_file /outputs/default_pipeline.yml --data_config_file /outputs/data_config_cpac.yml`

### Expected behavior

A default pipeline for preprocessing might be expected ... <span title="🙂">:)</span>

### Acceptance criteria

hopefully,,, any guidance or sugg

### C-PAC version

when I type cpac --version in the terminal, it says 'cpac (convenience wrapper) version 0.5.0 For C-PAC version, run `cpac version` with any cpac options (e.g., `--platform`, `--image`, `--tag`) that you would use while running'

### Container platform

Docker

### Docker and/or Singularity version(s)

Docker v.20.10.13

**Comment by suezaan** [comment]
Wow, thanks for your detailed answer. 
It really helped me understanding overall process! 

I am currently running preprocessing with what you have suggested, 
(add `-v /Users/hal/Desktop/test:/Users/hal/Desktop/test` command)
and for now it seems working though I have to take a look at the output in the end.

I hope you don't mind asking a few more questions...!

Regarding the problem that you wrote about cpac image packaging,
"The path /Users and its subdirectories don't exist in the C-PAC image as packaged, and nothing is being bound to those places in the image. Since loading the data is a very early step, it's hard to say if there are any other issues here or if that's causing all the trouble." 

does this seem related to the error that I got earlier when I tried to run command for creating data_settings file 
`cpac utils data_config new_settings_template`?

I got this error with this command:
`File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
OSError: [Errno 30] Read-only file system: '/logs'`

I ran it with docker cause it this didn't work for me, so I wonder if this is related to the problem that you said. 

Thank you so much for your concern! It really helped me a lot 👍

**Comment by shnizzedy** [comment]
`OSError: [Errno 30] Read-only file system: '/logs'` looks like this known issue: https://github.com/FCP-INDI/cpac/issues/21. I'm guessing you're running on MacOS?

**Comment by suezaan** [comment]
Exactly. I'm running on macOS Big Sur.

In fact, I am the one who posted this issue [[CPAC error in creating data_settings.yml template file](https://neurostars.org/t/cpac-error-in-creating-data-settings-yml-template-file/22824)](url) 

Good to see you again here!  

I couldn't figure out how to deal with the 'cpac utils' command and took a detour with docker command to create data_settings.yml. 

Guess that issue hasn't resolved yet, has it?

**Comment by shnizzedy** [comment]
https://github.com/FCP-INDI/cpac/issues/21 was resolved in [cpac 1.8.5](https://pypi.org/project/cpac/1.8.5/) ― `pip install --upgrade cpac` and you should be good to go!
